src/live_restore/live_restore_private.h

```diff
@@ -55,8 +55,31 @@ struct __wt_live_restore_file_handle {
     } destination;
 
     WT_FS_OPEN_FILE_TYPE file_type;
+    WT_RWLOCK ext_lock; /* File extent list lock */
 };
 
+/*
+ * WT_WITH_LIVE_RESTORE_EXTENT_LIST_READ_LOCK --
+ *     Acquire the extent list read lock and perform an operation.
+ */
+#define WT_WITH_LIVE_RESTORE_EXTENT_LIST_READ_LOCK(session, lr_fh, op) \
+    do {                                                               \
+        __wt_readlock((session), &(lr_fh)->ext_lock);                  \
+        op;                                                            \
+        __wt_readunlock((session), &(lr_fh)->ext_lock);                \
+    } while (0)
+
```

Comment:
The READ_LOCK version can go away if you're not going to use it (or keep it, and create a follow-on ticket and FIXME to wrap current read-lock uses to use it if you plan that for the future).

Reply:
188ad05a4